### ISSUE TITLE
define varible with string into Type Literal cause error

### ISSUE THREAD

**Issue by fxyu** [microsoft/pyright#6033]
## Environment data

-   Language Server version: v2023.9.20
-   OS and version: centos7
-   Python version (& distribution if applicable, e.g. Anaconda): python3.8.16

## Code Snippet

```python
from typing import Literal

class Status(object):
    ACTIVE = 'active'
    STOP = 'stop'
    
Literal[Status.ACTIVE, Status.STOP]
```
This cause the pylance report : Type arguments for "Literal" must be None, a literal value (int, bool, str, or bytes), or an enum value.

I used mypy directly to static check the file with no error.

![image](https://github.com/microsoft/pylance-release/assets/15837932/a7324ea0-497c-4c51-b3b0-7f393be2c64c)

**Comment by erictraut** [comment]
Pyright (the type checker upon which pylance is built) is correct in generating an error here. Mypy is allowing something that it should not.

[PEP 586](https://peps.python.org/pep-0586/#legal-parameters-for-literal-at-type-check-time), which introduces `Literal` to the Python type system, is very clear on what can and cannot appear within a `Literal` type annotation. In particular, you are not allowed to use variables in a `Literal` as you are attempting to do here, even if those variables have a literal type themselves.

You _are_ allowed to use type aliases that refer to other literal types, as in the following:

```python
class Status:
    ACTIVE = Literal["active"]
    STOP = Literal["stop"]

Literal[Status.ACTIVE, Status.STOP]
```

You are also allowed to use `enum` values, as in the following:

```python
class Status(Enum):
    ACTIVE = "active"
    STOP = "stop"

Literal[Status.ACTIVE, Status.STOP]
```
